modules/javafx.graphics/src/main/java/com/sun/javafx/tk/Toolkit.java

```diff
@@ -480,6 +489,11 @@ public void removePostSceneTkPulseListener(TKPulseListener listener) {
         }
     }
 
+    public void addCleanupListener(TKPulseListener listener) {
+        AccessControlContext acc = AccessController.getContext();
+        cleanupListeners.put(listener,acc);
```

Comment:
Access to `cleanupListeners` is not synchronized on `this` here, but it is synchronized in L426.
You should either synchronize each and every access, or none of it.

Reply:
I've added a synchronized now.